File: backend/app/src/main/java/com/onair/hearit/app/category/application/CategoryService.java
package com.onair.hearit.app.category.application;

import com.onair.hearit.app.common.dto.request.PagingRequest;
import com.onair.hearit.app.common.dto.response.PagedResponse;
import com.onair.hearit.app.category.dto.CategoryResponse;
import com.onair.hearit.core.domain.Category;
import com.onair.hearit.core.infrastructure.jpa.CategoryRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class CategoryService {

    private final CategoryRepository categoryRepository;

    public PagedResponse<CategoryResponse> getCategories(PagingRequest pagingRequest) {
        Pageable pageable = PageRequest.of(pagingRequest.page(), pagingRequest.size());
        Page<Category> categories = categoryRepository.findAll(pageable);
        Page<CategoryResponse> categoryDtos = categories.map(CategoryResponse::from);
        return PagedResponse.from(categoryDtos);
    }
}

File: android/app/src/main/java/com/onair/hearit/data/api/CategoryService.kt
package com.onair.hearit.data.api

import com.onair.hearit.data.dto.CategoryResponse
import com.onair.hearit.data.dto.SearchHearitResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface CategoryService {
    @GET("categories")
    suspend fun getCategories(
        @Query("page") page: Int?,
        @Query("size") size: Int?,
    ): Response<CategoryResponse>

    @GET("categories/{categoryId}/hearits")
    suspend fun getHearitsByCategoryId(
        @Path("categoryId") categoryId: Long,
        @Query("page") page: Int?,
        @Query("size") size: Int?,
    ): Response<SearchHearitResponse>
}
